lib/3scale/backend/stats/partition_eraser_job.rb

```diff
@@ -0,0 +1,63 @@
+module ThreeScale
+  module Backend
+    module Stats
+      # Job for deleting service stats
+      # Perform actual key deletion from a key partition definition
+      class PartitionEraserJob < BackgroundJob
+        # low priority queue
+        @queue = :stats
+
+        class << self
+          include StorageHelpers
+          include Configurable
+
+          def perform_logged(_enqueue_time, service_id, applications, metrics, users,
+                             from, to, offset, length, context_info = {})
+            job = DeleteJobDef.new(
+              service_id: service_id,
+              applications: applications,
+              metrics: metrics,
+              users: users,
+              from: from,
+              to: to
+            )
+
+            validate_job(job, offset, length)
+
+            stats_key_types = KeyTypesFactory.create(job)
+
+            stats_key_gen = KeyGenerator.new(stats_key_types)
```

Comment:
I think it would make sense to define an interface that returns the keys with one call instead of 2 like here.

Reply:
:+1: